geo/src/algorithm/convex_hull/qhull.rs

```diff
@@ -98,21 +99,36 @@ fn hull_set<T>(
             p_orth.x * p_diff.x + p_orth.y * p_diff.y
         })
         .enumerate()
-        .max_by(|(_, a), (_, b)| a.partial_cmp(b).unwrap())
-        .unwrap()
-        .0;
+        .try_fold(
+            (0, T::zero()),
+            |(farthest_idx, farthest_dist), (curr_idx, curr_dist)| {
+                // Ensure `distance` is not NaN
+                // TODO: we need a better check
```

Comment:
TODO: what about Infinity and -Infinity?

Reply:
My inclination would be to lean into the external semantics and percolate up an Err if and only if `a.partial_cmp(b)` is `None`.